samcli/lib/utils/file_observer.py

```diff
@@ -263,6 +299,7 @@ def __init__(self, on_change: Callable) -> None:
         self._images_observer_thread: Optional[Thread] = None
         self._lock: Lock = threading.Lock()
 
+    @broken_pipe_handler
```

Comment:
Not going to block on this. Might be even better if we can inject the behavior into the decorator.

```python
@os_exception_handler(platform="windows", exception=BROKEN_PIPE_ERROR)
```

Reply:
Interesting idea, though with this it might need a bit more rewriting/logic to be more generic. As of right now it has a pretty specific check for a particular error code vs a more generic callback/handler argument. I'll add a note in the code that this could be abstracted if something like this happens again